### PR TITLE
Support staking for new SDK

### LINKED ISSUE #681: Support staking again with the new SDK.
### Version

0.6.0 (once #591 merges)

### Description

Let's support staking again with the new SDK. I don't know how far along staking is in Cosmos SDK, we will have to ask. @mappum?

- [ ] support staking

### PR DESCRIPTION
**Pull request by mappum**
Closes #681 

Description:

The staking REST API is still in progress, but I updated the Voyager side as best as I could without testing against the actual LCD. This PR updates the LCD client and delegation/delegates modules, re-enables a lot of staking tests, restores staking functionality for the mock client, etc.

Maybe this should wait until the staking REST API changes are merged in the SDK repo (https://github.com/cosmos/cosmos-sdk/pull/994), but until then we can compile gaia from that branch when testing this.

### CODE REVIEW

**Comment by NodeGuy** [comment]
Shooting at a moving target!  🏹

**Comment by mappum** [comment]
😁 Woo

I'll test this against a local testnet on new SDK, butt maybe we shouldn't
merge this until a testnet exists based on this version of the SDK (the
REST handlers won't work with gaia-5001).

**Comment by faboweb** [comment]
How about a localnode with the current version?

**Comment by mappum** [comment]
This is working well now on gaia v0.18.0, not sure why the e2e tests failed on Circle.

**Comment by faboweb** [comment]
The CI points to an old version of gaia so the tests are expected to fail.

**Comment by faboweb** [comment]
Fixed e2e tests. @mappum do you take care of coverage? This moves us to build for 0.18.0 on next release. gaia-6000 is coming up soon though.

**Comment by mappum** [comment]
Sorry this has been drawn out for so long while being so close to done. Improved the coverage, now all that's left is a final review 😄

File: app/src/renderer/vuex/modules/delegation.js [hunk 1]
```diff
@@ -57,57 +57,73 @@ export default ({ commit, node }) => {
...
-              amount = { denom: "fermion", amount: Math.abs(amountChange) }
-            } else {
-              amount = Math.abs(amountChange)
+        if (isBond) {
+          delegate.push({
+            delegator_addr: rootState.wallet.address,
+            validator_addr: candidateId,
+            // TODO: figure out which denom is bonding denom
+            bond: {
+              denom: "steak",
```
Comment (faboweb) [inline app/src/renderer/vuex/modules/delegation.js]: Let's get this from some config?

**Comment by faboweb** [comment]
Merging as we need to upgrade to 0.18.0 anyway.

**Pull request merged**